script.js

```diff
@@ -1,4 +1,446 @@
+            playerCardsTemp[1].name == 10)) ||
+        (playerCardsTemp[1].name == "Ace" &&
+          (playerCardsTemp[0].name == "Jack" ||
+            playerCardsTemp[0].name == "Queen" ||
+            playerCardsTemp[0].name == "King" ||
+            playerCardsTemp[0].name == 10))
+      ) {
+        var cardPoints = 21;
+        cardNumberPerPlayer.push(cardPoints);
+      } else {
+        var cardIndex = 0;
+        var cardPoints = 0;
+
+        while (cardIndex < playerCardsTemp.length) {
+          cardPoints += playerCardsTemp[cardIndex].rank;
+          cardIndex += 1;
+        }
+
+        cardNumberPerPlayer.push(cardPoints);
+      }
+
+      var nextPlayer = (currentPlayer % numOfPlayers) + 1;
+
+      if (currentPlayer == numOfPlayers) {
+        currentPlayer = 1;
+        gameMode = GAME_MODE_LEADERBOARD;
+        return `Please click on "Submit" to see who won!`;
+      }
+
+      currentPlayer = nextPlayer;
+      gameMode = GAME_MODE_DECISION;
+      return `Hi ${
+        userName[currentPlayer - 1]
+      }, please click on "Submit" to view your cards!`;
+    }
+  } else if (gameMode == GAME_MODE_ACE_POST_DECISION) {
+    if (input == "hit") {
+      var card = newDeck.pop();
+      playerCardsTemp.push(card);
+
+      var cardIndex = 0;
+      var number = 1;
+      var standardOutput = "";
+
+      while (cardIndex < playerCardsTemp.length) {
+        standardOutput = `${standardOutput}(${number}) ${playerCardsTemp[cardIndex].name} of ${playerCardsTemp[cardIndex].suitPic}<br>`;
+
+        cardIndex += 1;
+        number += 1;
+      }
+
+      return `Hi ${
+        userName[currentPlayer - 1]
+      }, here are your cards:<br><br>${standardOutput}<br>Enter "hit" to draw a new card.<br>Enter number to finalize your hand.`;
+    } else if (Number(input)) {
+      if (Number.isNaN(Number(input)) == true) {
+        return `Please enter a number!`;
+      }
+
+      var cardPoints = Number(input);
+      cardNumberPerPlayer.push(cardPoints);
+
+      var nextPlayer = (currentPlayer % numOfPlayers) + 1;
+
+      if (currentPlayer == numOfPlayers) {
+        currentPlayer = 1;
+        gameMode = GAME_MODE_LEADERBOARD;
+        return `Please click on "Submit" to see who won!`;
+      }
+
+      currentPlayer = nextPlayer;
+      gameMode = GAME_MODE_DECISION;
+      return `Hi ${
+        userName[currentPlayer - 1]
+      }, please click on "Submit" to view your cards!`;
+    }
+  } else if (gameMode == GAME_MODE_LEADERBOARD) {
+    cardNumberPerPlayer = cardNumberPerPlayer.map(Number);
+    var returnIndexOfHighestNumber = Number(
+      indexOfMaxInArray(cardNumberPerPlayer)
+    );
+
+    if (cardNumberPerPlayer[returnIndexOfHighestNumber] <= 21) {
+      incrementNumberOfWinsOfEachPlayer(returnIndexOfHighestNumber);
+      countWagerPointSystem(returnIndexOfHighestNumber);
+
+      var playerWhoWon = returnIndexOfHighestNumber + 1;
+
+      var number = 1;
+      var index = 0;
+      var output = "";
```

Comment:
maybe we could have a more descriptive name that number and index as it is not very clear to me what these values should represent.